passive/clacks.js

```diff
@@ -1,35 +1,30 @@
 // Clacks Header Check by freakyclown@gmail.com
 
-function scan(ps, msg, src) {
-  var alertRisk = 0;
-  var alertConfidence = 3;
-  var alertTitle = "Server is running on CLACKS - GNU Terry Pratchett";
-  var alertDesc =
-    "The web/application server is running over the CLACKS network, some say its turtles/IP, some says its turtles all the way down the layer stack.";
-  var alertSolution =
-    "Give the sys admin a high five and rejoice in the disc world.";
+var ScanRuleMetadata = Java.type(
+  "org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata"
+);
 
-  var cweId = 200;
-  var wascId = 13;
+function getMetadata() {
+  return ScanRuleMetadata.fromYaml(`
+id: 100002
+name: Server is running on Clacks - GNU Terry Pratchett
+description: > 
+  The web/application server is running over the Clacks network, some say it's turtles/IP, 
+  some say it's turtles all the way down the layer stack.
+solution: Give the sysadmin a high five and rejoice in the disc world.
+references:
+  - https://xclacksoverhead.org/home/about
+risk: info
+confidence: high
+cweId: 200  # CWE-200: Exposure of Sensitive Information to an Unauthorized Actor
+wascId: 13  # WASC-13: Information Leakage
+status: alpha
+`);
+}
 
-  var url = msg.getRequestHeader().getURI().toString();
+function scan(helper, msg, src) {
   var headers = msg.getResponseHeader().getHeaders("X-Clacks-Overhead");
-
   if (headers != null) {
-    ps.raiseAlert(
-      alertRisk,
-      alertConfidence,
-      alertTitle,
-      alertDesc,
-      url,
-      "",
-      "",
-      "",
-      alertSolution,
-      headers,
-      cweId,
-      wascId,
-      msg
-    );
+    helper.newAlert().setMessage(msg).raise();
```

Comment:
Probably doesn't need to set message? I believe elsewhere @thc202 mentioned that the current message is set by default.

Reply:
That's for the URL.

Reply:
Oh okay, ignore my comments then